subprojects/packagefiles/libssh2/meson.build

```diff
@@ -0,0 +1,144 @@
+        libmbedx509 = cc.find_library('mbedx509')
+        libmbedcrypto = cc.find_library('mbedcrypto')
+        libssh2_deps += [libmbedtls, libmbedx509, libmbedcrypto]
+    endif
+
+    extra_flags += ['-DLIBSSH2_MBEDTLS']
+elif get_option('crypto_backend') == 'gcrypt'
+    libssh2_sources += ['src/libgcrypt.c']
+    libssh2_deps += dependency('gcrypt')
+    extra_flags += ['-DLIBSSH2_GCRYPT']
+else
+    error('No crypto backend specified')
+endif
+
+libssh2_config = configuration_data()
+
+check_lib_headers = [
+    ['unistd.h', 'HAVE_UNISTD_H'],
+    ['inttypes.h', 'HAVE_INTTYPES_H'],
+    ['stdlib.h', 'HAVE_STDLIB_H'],
+    ['sys/select.h', 'HAVE_SYS_SELECT_H'],
+    ['sys/uio.h', 'HAVE_SYS_UIO_H'],
+    ['sys/socket.h', 'HAVE_SYS_SOCKET_H'],
+    ['netinet/in.h', 'HAVE_NETINET_IN_H'],
+    ['arpa/inet.h', 'HAVE_ARPA_INET_H'],
+    ['sys/ioctl.h', 'HAVE_SYS_IOCTL_H'],
+    ['sys/time.h', 'HAVE_SYS_TIME_H'],
+    ['sys/un.h', 'HAVE_SYS_UN_H'],
+    ['windows.h', 'HAVE_WINDOWS_H'],
+    ['ws2tcpip.h', 'HAVE_WS2TCPIP_H'],
+    ['winsock2.h', 'HAVE_WINSOCK2_H'],
+    ['ntdef.h', 'HAVE_NTDEF_H'],
+    ['ntstatus.h', 'HAVE_NTSTATUS_H']
+]
+
+foreach h : check_lib_headers
+    libssh2_config.set(h[1], cc.has_header(h[0]))
+endforeach
+
+if cc.has_header('fcntl.h')
+    libssh2_config.set('HAVE_O_NONBLOCK', cc.has_header_symbol('fcntl.h', 'O_NONBLOCK'))
+endif
+
+if cc.has_header('sys/ioctl.h')
+    libssh2_config.set('HAVE_FIONBIO', cc.has_header_symbol('ioctl.h', 'FIONBIO'))
+endif
+
+libssh2_config.set('HAVE_EVP_AES_128_CTR', cc.has_function('EVP_aes_128_ctr'))
+libssh2_config.set('HAVE_GETTIMEOFDAY', cc.has_function('gettimeofday'))
+libssh2_config.set('HAVE_POLL', cc.has_function('poll'))
+libssh2_config.set('HAVE_SELECT', cc.has_function('select'))
+libssh2_config.set('HAVE_SOCKET', cc.has_function('socket'))
+libssh2_config.set('HAVE_STRTOLL', cc.has_function('strtoll'))
+libssh2_config.set('HAVE_STRTOI64', cc.has_function('strtoi64'))
+libssh2_config.set('HAVE_SNPRINTF', cc.has_function('sprintf'))
+libssh2_config.set('HAVE_LONGLONG', 1)
+libssh2_config.set('LIBSSH2DEBUG', get_option('buildtype') == 'debug')
+
+libssh2_config_h = configure_file(
+    input: 'src/libssh2_config_cmake.h.in',
+    output: 'libssh2_config.h',
+    configuration: libssh2_config,
+    format: 'cmake',
+    install: true,
+    install_dir: 'include/'
+)
+libssh2_sources += [libssh2_config_h]
+
+inc_dir = include_directories(['include', '.'])
+libssh2_lib = library('ssh2',
+    libssh2_sources,
+    include_directories: inc_dir,
+    dependencies: libssh2_deps,
+    c_args: extra_flags,
+    install: true,
+    soversion: '1.0.1',
+    version: meson.project_version()
+)
+libssh2_dep = declare_dependency(
+    include_directories: inc_dir,
+    link_with: libssh2_lib
+)
+
+install_headers(
+    files('include/libssh2.h',
+          'include/libssh2_publickey.h',
+          'include/libssh2_sftp.h'
+    )
+)
+
+pkgconfig.generate(libssh2_lib, name: 'libssh2')
```

Comment:
description would be nice. otherwise meson does funny things with it, especially when built as a subproject.